enumerable.rb

```diff
@@ -0,0 +1,149 @@
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < results.length
+        yield(results[i])
+        i += 1
+      end
+      results
+    else
+      results.to_enum
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < results.length
+        yield(results[i], i)
+        i += 1
+      end
+      results
+    else
+      results.to_enum
+    end
+  end
+
+  def my_select
+    if block_given?
+      choice = []
+      results.my_each do |x|
+        if yield(x)
+          choice.push(x)
+        end
+      end
+      choice
+    else
+      results.to_enum
+    end
+  end
```

Comment:
- [ ] returns an array containing all elements of enum for which the given block returns a true value